flatliners/prometheusendpoint.py

```diff
@@ -0,0 +1,57 @@
+from time import sleep
+from collections import defaultdict
+from .baseflatliner import BaseFlatliner
+
+from prometheus_client import Gauge, start_http_server
+
+# Set up logging
+import logging
+_LOGGER = logging.getLogger(__name__)
+
+class PrometheusEndpoint(BaseFlatliner):
+    """docstring for PrometheusEndpoint."""
+    def __init__(self):
+        super().__init__()
+        _LOGGER.info("Prometheus Endpoint initialized.")
+        self.update_interval = 5*60 # seconds # This should be the scraping interval for your prometheus
+        self.metric_buffer = defaultdict(list)
+        self.weirdness_score_gauge = Gauge('weirdness_score','Weirdness score for the given Cluster and Version',['cluster','version'])
+
+    def on_next(self, x):
+        try:
+            self.metric_buffer[str(x.cluster)] = [str(x.version), x.weirdness_score, False]
```

Comment:
again a buffer?

why not 

```python
def on_next(self, x):
  self.weirdness_score_gauge.labels(cluster=x.cluster, version=x.version).set(x.weirdness_score)
  self.published[x.cluster] = time.now()
```

And in a loop you delete all metrics that have `published + 60minutes < time.now()`

Reply:
I was trying something like cache invalidation. Invalidate the values that have been published and delete them in the next loop. 
And every published value stays published for at least 5 minutes (the scraping interval).

I will implement what you are suggesting and test it, I don't see any reason this wouldn't work

Reply:
great. The problem I have with the buffer is, that it a) creates more complexity and b) we'll store the values multiple times. Once in the gauge object itself and also in the buffer....